--- origsrc/nginx-1.0.14/auto/lib/perl/conf	2012-02-05 14:02:59.000000000 -0600
+++ src/nginx-1.0.14/auto/lib/perl/conf	2012-03-16 05:54:40.708468000 -0500
@@ -42,8 +42,7 @@ if test -n "$NGX_PERL_VER"; then
         ngx_perl_ldopts=`echo $ngx_perl_ldopts | sed 's/ -pthread//'`
     fi
 
-    CORE_LINK="$CORE_LINK $ngx_perl_ldopts"
-    LINK_DEPS="$LINK_DEPS $NGX_OBJS/src/http/modules/perl/blib/arch/auto/nginx/nginx.so"
+    CORE_LINK="$CORE_LINK $ngx_perl_ldopts -Wl,--out-implib,$NGX_OBJS/libnginx.a"
 
     if test -n "$NGX_PERL_MODULES"; then
         have=NGX_PERL_MODULES value="(u_char *) \"$NGX_PERL_MODULES\""
--- origsrc/nginx-1.0.14/auto/lib/perl/make	2012-03-05 06:58:10.000000000 -0600
+++ src/nginx-1.0.14/auto/lib/perl/make	2012-03-16 06:12:43.610406500 -0500
@@ -5,10 +5,11 @@
 
 cat << END                                                    >> $NGX_MAKEFILE
 
-$NGX_OBJS/src/http/modules/perl/blib/arch/auto/nginx/nginx.so:		\
+$NGX_OBJS/src/http/modules/perl/blib/arch/auto/nginx/nginx.dll:		\
 		src/http/modules/perl/nginx.pm				\
 		src/http/modules/perl/nginx.xs				\
 		src/http/modules/perl/ngx_http_perl_module.h		\
+		$NGX_OBJS/nginx						\
 		$NGX_OBJS/src/http/modules/perl/Makefile
 	cp -p src/http/modules/perl/nginx.* $NGX_OBJS/src/http/modules/perl/
 
@@ -31,6 +32,8 @@ $NGX_OBJS/src/http/modules/perl/Makefile
 			NGX_OPENSSL=$OPENSSL				\
 		$NGX_PERL Makefile.PL					\
 			LIB=$NGX_PERL_MODULES				\
+			INSTALLDIRS=vendor				\
+			OPTIMIZE="\$(CFLAGS)"				\
 			INSTALLSITEMAN3DIR=$NGX_PERL_MODULES_MAN
 
 END
--- origsrc/nginx-1.0.14/src/http/modules/perl/Makefile.PL	2012-03-05 06:58:10.000000000 -0600
+++ src/nginx-1.0.14/src/http/modules/perl/Makefile.PL	2012-03-16 06:03:54.006114900 -0500
@@ -14,6 +14,8 @@ WriteMakefile(
     AUTHOR            => 'Igor Sysoev',
 
     CCFLAGS           => "$ENV{NGX_PM_CFLAGS}",
+
+    LIBS              => "-L../../../.. -lnginx",
     OPTIMIZE          => '-O',
 
     INC               => "-I ../../../../../src/core " .
