--- origsrc/nginx-1.10.0/auto/lib/openssl/conf	2016-04-26 08:31:19.000000000 -0500
+++ src/nginx-1.10.0/auto/lib/openssl/conf	2016-04-27 11:54:00.934540100 -0500
@@ -107,6 +107,7 @@ else
             have=NGX_SSL . auto/have
             CORE_INCS="$CORE_INCS $ngx_feature_path"
             CORE_LIBS="$CORE_LIBS $ngx_feature_libs"
+            NGX_LIB_OPENSSL="$ngx_feature_libs"
             OPENSSL=YES
         fi
     fi
--- origsrc/nginx-1.10.0/auto/module	2016-04-26 08:31:19.000000000 -0500
+++ src/nginx-1.10.0/auto/module	2016-04-27 12:11:44.770130000 -0500
@@ -40,7 +40,7 @@ if [ "$ngx_module_link" = DYNAMIC ]; the
     do
         case $lib in
 
-            LIBXSLT | LIBGD | GEOIP | PERL)
+            LIBXSLT | LIBGD | GEOIP | PERL | OPENSSL)
                 libs="$libs \$NGX_LIB_$lib"
 
                 if eval [ "\$USE_${lib}" = NO ] ; then
@@ -48,7 +48,7 @@ if [ "$ngx_module_link" = DYNAMIC ]; the
                 fi
             ;;
 
-            PCRE | OPENSSL | MD5 | SHA1 | ZLIB)
+            PCRE | MD5 | SHA1 | ZLIB)
                 eval USE_${lib}=YES
             ;;
 
--- origsrc/nginx-1.10.0/auto/modules	2016-04-26 08:31:19.000000000 -0500
+++ src/nginx-1.10.0/auto/modules	2016-04-27 12:20:19.506493300 -0500
@@ -1171,7 +1171,7 @@ if [ $MAIL != NO ]; then
         ngx_module_incs=
         ngx_module_deps=$MAIL_DEPS
         ngx_module_srcs=$MAIL_SRCS
-        ngx_module_libs=
+        ngx_module_libs=OPENSSL
         ngx_module_link=DYNAMIC
 
         . auto/module
@@ -1191,7 +1191,7 @@ if [ $STREAM != NO ]; then
         ngx_module_incs=
         ngx_module_deps=$STREAM_DEPS
         ngx_module_srcs=$STREAM_SRCS
-        ngx_module_libs=
+        ngx_module_libs=OPENSSL
         ngx_module_link=DYNAMIC
 
         . auto/module
