--- origsrc/nginx-1.4.3/auto/unix	2013-10-08 07:07:14.000000000 -0500
+++ src/nginx-1.4.3/auto/unix	2013-10-13 12:20:59.289035800 -0500
@@ -151,7 +151,7 @@ fi
 ngx_feature="crypt()"
 ngx_feature_name=
 ngx_feature_run=no
-ngx_feature_incs=
+ngx_feature_incs="#include <crypt.h>"
 ngx_feature_path=
 ngx_feature_libs=
 ngx_feature_test="crypt(\"test\", \"salt\");"
@@ -163,7 +163,7 @@ if [ $ngx_found = no ]; then
     ngx_feature="crypt() in libcrypt"
     ngx_feature_name=
     ngx_feature_run=no
-    ngx_feature_incs=
+    ngx_feature_incs="#include <crypt.h>"
     ngx_feature_path=
     ngx_feature_libs=-lcrypt
     . auto/feature
